be/src/formats/disk_range.h

```diff
@@ -0,0 +1,49 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+#include <cstdint>
+#include <vector>
+
+namespace starrocks {
+
+class DiskRange {
+public:
+    DiskRange(const int64_t off, const int64_t len);
+
+    /**
+    * Returns the minimal DiskRange that encloses both this DiskRange
+    * and otherDiskRange. If there was a gap between the ranges the
+    * new range will cover that gap.
+    */
+    DiskRange span(const DiskRange& other_disk_range) const;
+
+    int64_t get_end() const;
+
+    int64_t get_offset() const;
+
+    int64_t get_length() const;
+
+private:
+    int64_t _offset = 0;
+    int64_t _length = 0;
+};
+
+class DiskRangeHelper {
+public:
+    static void merge_adjacent_disk_ranges(std::vector<DiskRange> disk_ranges, const int64_t max_merge_distance,
```

Comment:
```suggestion
    static void merge_adjacent_disk_ranges(const std::vector<DiskRange>& disk_ranges, const int64_t max_merge_distance,
```
The input should be a reference to reduce the copy.